databend-jdbc/src/main/java/com/databend/jdbc/DatabendNodes.java

```diff
@@ -34,6 +66,61 @@ public DatabendClientLoadBalancingPolicy getPolicy() {
         return policy;
     }
 
+    @Override
+    public boolean discoverUris(OkHttpClient client, ClientSettings settings) {
+        // do nothing if discovery interval is not reached
+        Long lastDiscoveryTime = this.lastDiscoveryTime.get();
+        if (System.currentTimeMillis() - lastDiscoveryTime < discoveryInterval) {
+            return false;
+        }
+        List<URI> current_nodes = query_nodes_uris.get();
+        if (!this.lastDiscoveryTime.compareAndSet(lastDiscoveryTime, System.currentTimeMillis())) {
+            return false;
+        }
+
+        List<DiscoveryNode> new_nodes = DatabendClientV1.dicoverNodes(client, settings);
+        if (!new_nodes.isEmpty()) {
+            // convert new nodes using lambda
+            List<URI> new_uris = new_nodes.stream().map(node -> URI.create("http://" + node.getAddress())).collect(Collectors.toList());
+            updateNodes(new_uris);
+            return true;
+        }
+        return false;
+    }
+
+    private List<URI> parseURI(List<DiscoveryNode> nodes) throws SQLException {
```

Comment:
Add some UT for ` discoverUris` and `parseURI` function?

Reply:
I will write another PR fully test on those auto discovery related functions.